v0/pipeline.json

```diff
@@ -163,7 +163,8 @@
         "type" : "object",
         "properties" : {
           "type" : {
-            "type" : "string"
+            "type" : "string",
+            "enum" : [ "Slack", "Email", "PagerDuty", "MsTeams", "Webhook", "Datadog" ]
```

Comment:
Any specific reason we are introducing the enum now?

Reply:
This would help in YAML auto-complete.
Also give error in yaml-editor if wrong type is selected.

This would have been a breaking change but deserialisation itself fails while saving and running the pipeline. So we are sure that no there is no pipeline with invalid type. And if there is any in git, it can not run now as well.

Reply:
Thanks for explanation!